src/steps/data_validation/utils/classification_dataset_validator.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+from src.steps.data_validation.utils.dataset_validator import DatasetValidator
+
+
+class ClassificationDatasetValidator(DatasetValidator):
+    def validate(self):
+        super().validate()  # Call common validations
+        self.validate_labelmap()
+        self.validate_at_least_one_image_per_class()
+
+    def validate_labelmap(self):
+        for dataset_context in self.dataset_collection:
+            if len(dataset_context.labelmap) < 2:
+                raise ValueError(
+                    f"Labelmap must have at least 2 classes in {dataset_context.dataset_name} dataset"
+                )
+
+    def validate_at_least_one_image_per_class(self):
```

Comment:
Missing docstring, also why do we perform this check only on the train dataset? Shouldn't we validate the `val` and `test` splits as well?

Reply:
If the validation and test datasets are not well-represented across all classes, it's not necessarily our responsibility to ensure they are, or at least it shouldn't be a blocking issue. We can maybe implement these checks as warnings. Do you think it should be blocking ?

Reply:
Hum, I guess this should be blocking :
- What if, as a user, I did a mistake and did not configured my datasets correctly ?
- What if the split has created an empty dataset split (even if almost impossible) ?